private/manageFishData.py

```diff
@@ -56,64 +69,92 @@ def omit_tag_decoder(i, t, l):
 
     return coll
 
-XIV = load_dats()
-fish_and_tackle_data = {}
-
 
 def _is_town_or_field_territory(name):
     return len(name) == 4 and name[2] in ('f', 't', 'h') and str(name[3]).isdigit()
 
-TERRITORIES = list(filter(lambda data: data.get_raw('Map') != 0 and
-                                       _is_town_or_field_territory(data['Name']),
-                          XIV.get_sheet('TerritoryType')))
-
 
 def _collect_weather_rates(rate):
     return [(r[1].key, r[0]) for r in rate.weather_rates if r[1].key != 0]
 
-# Determine "useful" weather types.
-WEATHER_RATES = dict([
-    (territory.key,
-     OrderedDict({'map_id': territory.map.key,
-                  'zone_id': territory.place_name.key,
-                  'zone_name': str(territory.place_name.name),
-                  'region_id': territory.region_place_name.key,
-                  'region_name': str(territory.region_place_name.name),
-                  'weather_rates': _collect_weather_rates(territory.weather_rate)}))
-    for territory in TERRITORIES])
-
-WEATHER_TYPES = dict([
-    (weather.key,
-     OrderedDict({'name': str(weather.name),
-                  'icon': '%06u' % weather.get_raw('Icon')}))
-    for weather in
-    set(reduce(add, [t.weather_rate.possible_weathers for t in TERRITORIES], []))
-    if weather.key != 0])
-
-FISHING_NODES = dict([
-    (spot.key,
-     OrderedDict({'_id': spot.key,
-                  'name': str(spot.as_string('PlaceName')),
-                  'territory_id': spot.get_raw('TerritoryType')}))
-    for spot in XIV.get_sheet('FishingSpot')])
-
-SPEARFISHING_NODES = dict([
-    (x['GatheringPointBase'].key,
-     OrderedDict({'_id': x['GatheringPointBase'].key,
-                  'name': str(x.as_string('PlaceName')) if x.get_raw('PlaceName') != 0 else 'Node',
-                  'territory_id': x.get_raw('TerritoryType')}))
-    for x in XIV.get_sheet('GatheringPoint')
-    if x['GatheringPointBase']['GatheringType'].key == 4])
-
-# Store the dictionaries sorted by key
-# This makes the generated JS a bit more consistent.
-FISHING_NODES = OrderedDict(sorted(FISHING_NODES.items(), key=lambda t: t[0]))
-SPEARFISHING_NODES = OrderedDict(sorted(SPEARFISHING_NODES.items(), key=lambda t: t[0]))
-WEATHER_RATES = OrderedDict(sorted(WEATHER_RATES.items(), key=lambda t: t[0]))
-WEATHER_TYPES = OrderedDict(sorted(WEATHER_TYPES.items(), key=lambda t: t[0]))
-
-KeyValuePair = namedtuple('KeyValuePair', ['key', 'value'])
 
+def initialize_data(args):
+  global XIV
+  global KeyValuePair
+  global FISHING_NODES
+  global SPEARFISHING_NODES
+  global WEATHER_RATES
+  global WEATHER_TYPES
+  global ICON_MAP
+
+  XIV = load_dats(args)
+
+  TERRITORIES = list(
+                  filter(
+                    lambda data: data.get_raw('Map') != 0 and _is_town_or_field_territory(data['Name']), XIV.get_sheet('TerritoryType')
```

Comment:
Reformat this line so it's not super long.  The way it was previously formatted should be fine.